frontend/src/scenes/experiments/utils.test.ts

```diff
@@ -0,0 +1,82 @@
+import { EntityType, InsightType } from '~/types'
+
+import { getMinimumDetectableEffect } from './utils'
+
+describe('utils', () => {
+    it('Funnel experiment returns correct MDE', async () => {
+        const experimentInsightType = InsightType.FUNNELS
+        const trendResults = [
+            {
+                action: {
+                    id: '$pageview',
+                    type: 'events' as EntityType,
+                    order: 0,
+                    name: '$pageview',
+                    custom_name: null,
+                    math: 'total',
+                    math_group_type_index: null,
+                },
+                aggregated_value: 0,
+                label: '$pageview',
+                count: 0,
+                data: [],
+                labels: [],
+                days: [],
+            },
+        ]
+
+        let conversionMetrics = { averageTime: 0, stepRate: 0, totalRate: 0.01 }
+        expect(getMinimumDetectableEffect(experimentInsightType, conversionMetrics, trendResults)).toEqual(1)
+
+        conversionMetrics = { averageTime: 0, stepRate: 0, totalRate: 0.1 }
+        expect(getMinimumDetectableEffect(experimentInsightType, conversionMetrics, trendResults)).toEqual(5)
+
+        conversionMetrics = { averageTime: 0, stepRate: 0, totalRate: 0.3 }
+        expect(getMinimumDetectableEffect(experimentInsightType, conversionMetrics, trendResults)).toEqual(3)
+
+        conversionMetrics = { averageTime: 0, stepRate: 0, totalRate: 0.5 }
+        expect(getMinimumDetectableEffect(experimentInsightType, conversionMetrics, trendResults)).toEqual(6)
```

Comment:
Because JS does this to numbers 😭 :
55 * 1.1 = 55.00000000000001

And then we're rounding the MDE up, so that for small conversion rates the MDE is always at least 1.

But good catch, fixed 👍

Reply:
oh lol 👍